design/aws-ecs-autoscaling-queue-worker.md

```diff
@@ -0,0 +1,225 @@
+# AWS ECS - L3 Construct for Autoscaling ECS/Fargate Service that Processes Items in a SQS Queue
+
+To address issue [#2396](https://github.com/awslabs/aws-cdk/issues/2396), the AWS ECS CDK construct library should provide a way for customers to create a Queue Worker service (an AWS ECS/Fargate Service that processes items from an sqs queue). This construct will create an SQS Queue and enable autoscaling for the service based on cpu usage and the sqs approximateNumberOfMessagesVisible metric. This would mean adding new ECS CDK constructs `Ec2QueueWorkerService` and `FargateQueryWorkerService`, that would take in the necessary properties required to create a Task Definition, an SQS Queue as well as an ECS/Fargate Service with Auto Scaling rules.
```

Comment:
Style nit: This paragraph is a little confusingly worded -- there is mention of "this construct" before the new constructs are named. Also some wonky spacing/inconsistent capitalization?

Reply:
Updated